### PR TITLE
finished CCamera

### PR DESCRIPTION
**Pull request by aap**

### CODE REVIEW

File: src/core/Camera.cpp [hunk 1]
```diff
@@ -4,137 +4,3079 @@
...
+	m_bRestoreByJumpCut = false;
+	CarZoomIndicator = 2.0f;
+	PedZoomIndicator = 2.0f;
+	CarZoomValueSmooth = 0.0f;
+	m_fPedZoomValueSmooth = 0.0f;
+	pTargetEntity = nil;
+	if(FindPlayerVehicle())
+		pTargetEntity = FindPlayerVehicle();
+	else
+		pTargetEntity = CWorld::Players[CWorld::PlayerInFocus].m_pPed;
```
Comment (erorcun) [inline src/core/Camera.cpp:135]: FindPlayerPed()

File: src/core/Camera.cpp [hunk 2]
```diff
@@ -4,137 +4,3079 @@
...
+				// Car zoom value
+				if(CarZoomIndicator == CAM_ZOOM_1STPRS && !m_bPlayerIsInGarage){
+					CarZoomValue = 0.0f;
+					ReqMode = CCam::MODE_1STPERSON;
+				}else if(CarZoomIndicator == CAM_ZOOM_1)
+					CarZoomValue = 0.05f;
+				else if(CarZoomIndicator == CAM_ZOOM_2)
+					CarZoomValue = 1.9f;
+				else if(CarZoomIndicator == CAM_ZOOM_3)
+					CarZoomValue = 3.0f;
```
Comment (erorcun) [inline src/core/Camera.cpp:735]: 3.9f

File: src/core/Camera.cpp [hunk 3]
```diff
@@ -4,137 +4,3079 @@
...
+					m_fPedZoomValueSmooth -= 0.12f * CTimer::GetTimeStep();
+					m_fPedZoomValueSmooth = max(m_fPedZoomValueSmooth, m_fPedZoomValue);
+				}
+			}
+
+			WellBufferMe(CloseInPedHeightTarget, &Cams[ActiveCam].m_fCloseInPedHeightOffset, &Cams[ActiveCam].m_fCloseInPedHeightOffsetSpeed, 0.1f, 0.025f, false);
+
+			// Check if entering fight cam
+			if(!m_bFirstPersonBeingUsed){
+				if(FindPlayerPed()->GetPedState() == PED_ATTACK && !m_bUseMouse3rdPerson)
```
Comment (erorcun) [inline src/core/Camera.cpp:891]: PED_FIGHT

File: src/core/Camera.cpp [hunk 4]
```diff
@@ -4,137 +4,3079 @@
...
+					ReqMode = CCam::MODE_FIGHT_CAM;
+				if(((CPed*)pTargetEntity)->GetWeapon()->m_eWeaponType == WEAPONTYPE_BASEBALLBAT &&
+				   FindPlayerPed()->GetPedState() == PED_PAUSE && !m_bUseMouse3rdPerson)
```
Comment (erorcun) [inline src/core/Camera.cpp:894]: PED_ATTACK .d

File: src/core/Camera.cpp [hunk 5]
```diff
@@ -4,137 +4,3079 @@
...
+						garageCenterToDoor.Normalise();
+						if(whichDoor == 1){
+							if(pToGarageWeAreIn == nil && stairs)
+								garageCamPos = garageDoorPos1 + 3.75f*garageCenterToDoor;
+							else
+								garageCamPos = garageDoorPos1 + 13.0f*garageCenterToDoor;
+						}else{
+							garageCamPos = garageDoorPos2 + 13.0f*garageCenterToDoor;
+						}
+						if(PedZoomIndicator == CAM_ZOOM_TOPDOWN && stairs){
```
Comment (erorcun) [inline src/core/Camera.cpp:976]: && !stairs)

File: src/core/Camera.cpp [hunk 6]
```diff
@@ -4,137 +4,3079 @@
...
+							garageCamPos = garageCenter;
+							garageCamPos.z += 3.1f*FindPlayerPed()->GetPosition().z;
```
Comment (erorcun) [inline src/core/Camera.cpp:978]: garageCamPos.z += 2.1f **+** FindPlayerPed()->GetPosition().z;

File: src/core/Camera.cpp [hunk 7]
```diff
@@ -4,137 +4,3079 @@
...
+					bool targetDead = false;
+					float camAngle, targetAngle;
+					CVector playerToTarget = m_cvecAimingTargetCoors - pTargetEntity->GetPosition();
+					CVector playerToCam = Cams[ActiveCam].Source - pTargetEntity->GetPosition();
+
+					if(PedZoomIndicator == CAM_ZOOM_1)
+						deadPedDist = 2.25f;
+					if(FindPlayerPed()->m_pPointGunAt){
+						// BUG: this need not be a ped!
+						if(((CPed*)FindPlayerPed()->m_pPointGunAt)->GetPedState() == PED_DEAD ||
```
Comment (erorcun) [inline src/core/Camera.cpp:1056]: or ->DyingOrDead()

File: src/core/Camera.h [hunk 8]
```diff
@@ -4,13 +4,28 @@
...
+	MAX_NUM_OF_NODES = 800	// for trains
+};
+
+#define DEFAULT_NEAR (0.9f)
+#define CAM_ZOOM_1STPRS (0.0f)
+#define CAM_ZOOM_1 (1.0f)
+#define CAM_ZOOM_2 (2.0f)
+#define CAM_ZOOM_3 (3.0f)
+#define CAM_ZOOM_TOPDOWN (4.0f)
+#define CAM_ZOOM_CINEMATIC (5.0f)
```
Comment (Sergeanur) [inline src/core/Camera.h:24]: Why do this instead of fixing types for CarZoomIndicator and PedZoomIndicator?

Comment (aap) [reply src/core/Camera.h:24]: what do you mean? make them ints? the game uses floats and i would like to keep it that way.

Comment (Sergeanur) [reply src/core/Camera.h:24]: Make them ints, yeah. Use of floats was clearly a copypaste mistake fixed in SA.

**Pull request merged**